JavaScript/JavaScriptSDK/Sender.ts

```diff
@@ -27,6 +27,65 @@ declare var XDomainRequest: {
 module Microsoft.ApplicationInsights {
     "use strict";
 
+    export class DataLossAnalyzer {
+        static enabled = false;
+        static appInsights: Microsoft.ApplicationInsights.AppInsights;
+
+        static reset() {
+            sessionStorage.setItem("itemsQueued", "0");
+        }
+
+        static itemQueued() {
+            try {
+                if (sessionStorage && sessionStorage.getItem && sessionStorage.setItem) {
+                    var itemsQueued = sessionStorage.getItem("itemsQueued");
+                    itemsQueued = itemsQueued || 0;
+                    ++itemsQueued;
+                    sessionStorage.setItem("itemsQueued", itemsQueued);
+                }
+            } catch (e) { }
+        }
+
+        static itemsSentSuccessfully(countOfItemsSentSuccessfully: number) {
+            try {
+                if (sessionStorage && sessionStorage.getItem && sessionStorage.setItem) {
+                    var itemsQueued = sessionStorage.getItem("itemsQueued");
+                    itemsQueued = itemsQueued
+                        ? (itemsQueued - countOfItemsSentSuccessfully)
+                        : 0;
+                    sessionStorage.setItem("itemsQueued", itemsQueued);
+                }
+            } catch (e) { }
+        }
+
+        static getNumberOfLostItems(): number {
+            try {
+                if (sessionStorage && sessionStorage.getItem && sessionStorage.setItem) {
+                    var itemsQueued = sessionStorage.getItem("itemsQueued");
+                    itemsQueued = itemsQueued || 0;
+                    return itemsQueued;
+                }
+            } catch (e) {
+                return 0;
+            }
+        }
+
+        static reportLostItems() {
+            try {
+                if (DataLossAnalyzer.enabled && DataLossAnalyzer.appInsights && DataLossAnalyzer.getNumberOfLostItems() > 0) {
+                    DataLossAnalyzer.appInsights.trackTrace(
+                        "AI (Internal): Not all telemetry items were sent from the previous page. Please set maxBatchSizeInBytes and/or maxBatchInterval to increase sending frequency. Count of missing items: "
+                        + DataLossAnalyzer.getNumberOfLostItems()
+                        , null);          
```

Comment:
I don't think we should be making this an action item for the customer to change maxBatchSizeInBytes and maxBatchInterval - if I saw this I would be frightened and slightly annoyed I should be tweaking it and it doesn't just work out of the box :) I think we should send it as internal info for our own analysis only, without the message.